rust/src/transformations/make_stable_expr/expr_sum/mod.rs

```diff
@@ -0,0 +1,347 @@
+/// | ----------------------------------------- |
+/// | `SymmetricDistance`                       |
+/// | `InsertDeleteDistance`                    |
+/// | `PartitionDistance<SymmetricDistance>`    |
+/// | `PartitionDistance<InsertDeleteDistance>` |
+///
+/// # Arguments
+/// * `input_domain` - ExprDomain
+/// * `input_metric` - valid selections shown in table above
+/// * `expr` - sum expression
+pub fn make_expr_sum<MS, MI, const P: usize>(
+    input_domain: ExprDomain,
+    input_metric: MS,
+    expr: Expr,
+) -> Fallible<Transformation<ExprDomain, ExprDomain, MS, LpDistance<P, Scalar>>>
+where
+    MS: 'static + Metric,
+    MI: 'static + DatasetOuterMetric,
+    MI::Distance: Clone,
+    (ExprDomain, MS): MetricSpace,
+    (ExprDomain, MI): MetricSpace,
+    Expr: StableExpr<MS, MI>,
+{
+    let Expr::Agg(AggExpr::Sum(prior_expr)) = expr else {
+        return fallible!(MakeTransformation, "expected function expression");
+    };
+
+    let t_prior = prior_expr.make_stable(input_domain, input_metric)?;
+    let (input_domain, input_metric) =
+        (t_prior.output_domain.clone(), t_prior.output_metric.clone());
+
+    let dtype = &input_domain.active_series()?.field.dtype;
+
+    // check that we are in a context where it is ok to break row-alignment
+    input_domain.context.break_alignment()?;
+
+    // build output domain
+    let mut output_domain = input_domain.clone();
+
+    // summation invalidates bounds on the active column
+    output_domain.active_series_mut()?.drop_bounds()?;
+
+    // we only care about the margin that matches the current grouping columns
+    let margin_id = input_domain.context.grouping_columns();
+    let input_margin = (output_domain.frame_domain.margins.remove(&margin_id))
+        .ok_or_else(|| err!(MakeTransformation, "failed to find margin {:?}", margin_id))?;
+
+    // Set the margins on the output domain to consist of only one margin: 1 row per group, with at most 1 record in each group.
+    let output_margin = input_margin.clone().with_max_partition_size(1);
+    output_domain.frame_domain.margins = HashMap::from_iter([(margin_id, output_margin)]);
+
+    fn stability_map<MI, const P: usize, TI>(
+        series_domain: SeriesDomain,
+        margin: Margin,
+    ) -> Fallible<StabilityMap<MI, LpDistance<P, Scalar>>>
+    where
+        MI: DatasetOuterMetric,
+        MI::Distance: Clone,
+        TI: Summand,
+        Scalar: From<TI::Sum>,
+        f64: InfCast<TI::Sum>,
+    {
+        let (l, u) = series_domain.atom_domain::<TI>()?.get_closed_bounds()?;
+        let (l, u) = (TI::Sum::neg_inf_cast(l)?, TI::Sum::inf_cast(u)?);
+
+        let public_info = margin
+            .public_info
+            .ok_or_else(|| err!(MakeTransformation, "keys must be public information"))?;
+
+        let max_size = usize::exact_int_cast(margin.max_partition_size.ok_or_else(|| {
+            err!(
+                MakeTransformation,
+                "must specify max_partition_len in margin"
```

Comment:
Change to `_size` to be consistent with the code above? or change both to `_len`?